test/login_external_test.rb

```diff
@@ -42,18 +42,47 @@ def request_session
     end
 
     describe "token is correct" do
-      let(:cookie_data){{ "foo" => "bar" }}
-      let(:token){ JWT.encode(cookie_data, shared_key, "HS256") }
+      let(:jwt_data){{ "foo" => "bar" }}
+      let(:token){ JWT.encode(jwt_data, shared_key, "HS256") }
 
-      it "writes cookies and redirects to dashboard" do
-        any_instance_of(Identity::LoginExternal) do |finalize|
-          mock(finalize).write_authentication_to_cookie(cookie_data)
+      describe "there is no active oauth authorization request" do
+        it "writes cookies and redirects to dashboard" do
+          any_instance_of(Identity::LoginExternal) do |finalize|
+            mock(finalize).write_authentication_to_cookie(jwt_data)
+          end
+
+          get "/login/external?token=#{token}"
+
+          assert_equal 302, last_response.status
+          assert_equal Identity::Config.dashboard_url,
+                       last_response.headers["Location"]
         end
+      end
 
-        get "/login/external?token=#{token}"
+      describe "the user has an active oauth authorization request" do
+        before do
+          stub_heroku_api
+        end
 
-        assert_equal 302, last_response.status
-        assert_equal Identity::Config.dashboard_url, last_response.headers["Location"]
+        let(:jwt_data) do
+          { access_token: { token: "abcd", expires_in: 9000 },
+            session: { id: "456" },
+            user: { id: "123" },
+            sso_entity: "initech" }
+        end
+
+        let(:session_data) do
+          { "authorize_params" => MultiJson.encode("client_id" => "dashboard") }
+        end
+
+        it "finishes the authorization" do
+          get "/login/external", { token: token },
+                                 { "rack.session" => session_data }
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.